Insights Account Discovery blog post (#13463)
pulumi/docs: content/blog/insights-cloud-account-discovery/index.md

<!--more-->

### Account Management

To get started, Insights Account Discovery provides a simple UI approach to setting up all of your infrastructure accounts to be scanned and read into the Insights supergraph. Once discovered, all of your infrastructure resources can then be seen in the Resource Explorer and accessed by Pulumi CoPilot and Resource Search, providing you the tools needed to deliver on your strategic infrastructure goals.

Account Discovery starts with the new Accounts page to provide a list of all accounts created along with the current status of the latest account scans. The Accounts page is the single place to go to manage all aspects of the Insights Account Discovery process.

{{< video title="Accounts" src="accounts.mp4" autoplay="true" loop="true" >}}

### Account Creation

> From this page you can create new Accounts to provide the configuration and credentials needed for Insights to regularly scan and manage your infrastructure. Once you create a new top level Account, Insights will automatically create child accounts for each region you specified to enable you to control the discovery behavior for each region separately.

Reviewer: regions are AWS specific. We may introduce the concept of account hierarchies (accounts can be organized in hierarchies to control the discovery behavior, and for visualization purposes), and then exemplify with AWS regions - which is our only example.

Author: I changed to:

From this page you can create new Accounts to provide the configuration and credentials needed for Insights to regularly scan and manage your infrastructure. Once you create a new top level Account, Insights will automatically create child accounts, based on the underlying platform model, for each group you enabled. These child accounts enable you to control the discovery behavior for each group separately. For example, AWS enables you to divide you infrastructure into regions and Insights will create separate child accounts for each region you specify.

{{% notes type="info" %}}

> Account Discovery leverages Pulumi ESC to enter and manage the credentials needed for the Discovery process to find and read all of the infrastructure resources. By using ESC to manage the credentials, it enables Insights to leverage the enterprise best practices for creating and managing applicaiton secrets that Pulumi ESC provides.

Author: Good update. I changed to your text.

# https://twitter.com/PulumiCorp/status/1755637618631405655

social:
    twitter:
    linkedin:

# See the blogging docs at https://github.com/pulumi/docs/blob/master/BLOGGING.md
# for details, and please remove these comments before submitting for review.
---

Enterprise cloud infrastructures are complex environments that are evolved over time and made up of thousands of different kinds of resources. Enabling customers to wrap their arms around this complexity and get a complete understanding of the scope and structure is the goal of the Pulumi Insights 2.0 product.

We are excited to announce the public preview of Insights Account Discovery that makes it easy for you to gain visibility of your entire infrastructure regardless of how it is managed. Insights provides the tools to find, group and drill into your resources as needed to make sense of all aspects of your cloud infrastructure. In addition Copilot AI gives you the ability to get answers about your infrastructure that can be challenging to get in other ways, dramatically simplifying the work needed to move your infrastructure forward.

<!--more-->

### Account Management

> To get started, Insights Account Discovery provides a simple UI approach to setting up all of your infrastructure accounts to be scanned and read into the Insights supergraph. Once discovered, all of your infrastructure resources can then be seen in the Resource Explorer and accessed by Pulumi CoPilot and Resource Search, providing you the tools needed to deliver on your strategic infrastructure goals.

Reviewer 2: ```suggestion
To get started, Insights Account Discovery provides a simple UI approach to setting up all of your infrastructure accounts to be scanned and read into the Insights supergraph. Once discovered, all of your infrastructure resources can then be seen in the Resource Explorer and accessed by Pulumi Copilot and Resource Search, providing you the tools needed to deliver on your strategic infrastructure goals.
```

{{% notes type="info" %}}
Account Discovery leverages Pulumi ESC to enter and manage the credentials needed for the Discovery process to find and read all of the infrastructure resources. By relying on ESC, Insights aligns with enterprise best practices for creating and handling application secrets.
{{% /notes %}}

{{< video title="Create Account" src="create-account.mp4" autoplay="true" loop="true" >}}

### Understanding Your Infrastructure

Once each of your Accounts have started scanning, your resources will be displayed in the Insights Resources Explorer along with all Infrastructure as Code resources that you have imported. The Resources Explorer and Resources Search enables you to ask questions and factor your resources into logical groupings that significantly improves the process for managing your infrastructure and getting answers to key questions needed for your projects. For example, you can ask the AI Resource Search to 'Find all public IP addresses' and any other question you might have about your infrastructure.

{{< video title="Discovered Resources" src="resources.mp4" autoplay="true" loop="true" >}}

> ### Conclusion

Reviewer 2: There are two areas I think we should speak to before concluding:

1/ Policy. This is a fast follow, right? Should we at least make mention of it? Or will we just blog about it separately when it's available?

2/ Pricing. Should we say, "Free while in public preview, with per-resource pricing coming in Q1 2025"? People are often nervous to use something if they don't know how it will be priced.

Author: I talked about Policy in the last line of the conclusion. However I created a new section called 'More to Come' and moved this into that section:

_We are excited about the potential of Pulumi Insights and are working hard to bring additional capabilities to continue to simplify the experience of managing today's complex cloud environments. Watch for additional features like Pulumi Crossguard Policy support for Discovered resources, along with additional platform support, and more, coming soon._

_Insights Account Discovery is free while in public preview, with per-resource pricing for Team, Enterprise and Business Critical tiers coming in Q1 2025._

> The addition of Account Discovery significantly expands the scope of Pulumi Insights. You can now leverage the capabilities of Insights 2.0, not just for your Pulumi IaC managed resources, but for all resources in your infrastructure regardless of how it is managed. With Pulumi Copilot and Resource Search, you are able to gain insights and ask questions about your infrastructure that would otherwise be challenging to answer, saving time and providing the critical information needed.

Reviewer 2: ```suggestion
The addition of Account Discovery significantly expands the scope of Pulumi Insights. You can now leverage the capabilities of Insights 2.0, not just for your Pulumi IaC managed resources, but for all resources in your infrastructure regardless of how they are managed. With Pulumi Copilot and Resource Search, you are able to gain insights and ask questions about your infrastructure that would otherwise be challenging to answer, saving time and providing the critical information needed.
```

Author: Updated per your suggestion

> For additional information about Pulumi Insights please refer to: [Pulumi Insights](https://www.pulumi.com/product/pulumi-insights/)

Reviewer 2: Should we have a CTA to sign-up? Should it link directly to the "Register a New Account" page?

Author: The challenge is that Insights is not available to the Individual tier so the CTA would not be relevant to a portion of the readers. We are going to add an upsell screen for individual users that click on Accounts in the left nav, but today Accounts is hidden for individual users.
